astroplan/periodic.py

```diff
@@ -0,0 +1,295 @@
+
+        .. warning::
+            Barycentric offsets are ignored in the current implementation.
+
+        Parameters
+        ----------
+        time : `~astropy.time.Time`
+            Time to evaluate
+
+        Returns
+        -------
+        in_eclipse : `~numpy.ndarray` or bool
+            `True` if ``time`` is during primary eclipse
+        """
+        phases = self.phase(time)
+        return ((phases < float(self.duration/self.period)/2) |
+                (phases > 1 - float(self.duration/self.period)/2))
+
+    def in_secondary_eclipse(self, time):
+        r"""
+        Returns `True` when ``time`` is during a secondary eclipse
+
+        If the eccentricity of the eclipsing system is non-zero, then we compute
+        the secondary eclipse time approximated to first order in eccentricity,
+        as described in Winn (2010) Equation 33 [1]_:
+
+        The time between the primary eclipse and secondary eclipse :math:`\delta t_c`
+        is given by :math:`\delta t_c \approx 0.5 \left (\frac{4}{\pi} e \cos{\omega \right)`,
+        where :math:`e` is the orbital eccentricity and :math:`\omega` is the
+        angle of periapsis.
+
+        .. warning::
+            This approximation for the secondary eclipse time is only accurate
+            when the orbital eccentricity is small; and barycentric offsets
+            are ignored in the current implementation.
+
+        Parameters
+        ----------
+        time : `~astropy.time.Time`
+            Time to evaluate
+
+        Returns
+        -------
+        in_eclipse : `~numpy.ndarray` or bool
+            `True` if ``time`` is during secondary eclipse
+
+        References
+        ----------
+        .. [1] Winn (2010) https://arxiv.org/abs/1001.2010
+        """
+        if self.eccentricity < 1e-5:
+            secondary_eclipse_phase = 0.5
+        else:
+            secondary_eclipse_phase = 0.5 * (1 + 4/np.pi * self.eccentricity *
+                                             np.cos(self.argument_of_periapsis))
+        phases = self.phase(time)
+        return ((phases < secondary_eclipse_phase + float(self.duration/self.period)/2) &
+                (phases > secondary_eclipse_phase - float(self.duration/self.period)/2))
+
+    def out_of_eclipse(self, time):
+        """
+        Returns `True` when ``time`` is not during primary or secondary eclipse.
+
+        .. warning::
+            Barycentric offsets are ignored in the current implementation.
+
+        Parameters
+        ----------
+        time : `~astropy.time.Time`
+            Time to evaluate
+
+        Returns
+        -------
+        in_eclipse : `~numpy.ndarray` or bool
+            `True` if ``time`` is not during primary or secondary eclipse
+        """
+        return np.logical_not(np.logical_or(self.in_primary_eclipse(time),
+                                            self.in_secondary_eclipse(time)))
+
+    def next_primary_eclipse_time(self, time, n_eclipses=1):
```

Comment:
What about `next_eclipse_time_primary`?  Mostly for consistent tab completion, really...

Reply:
I understand the tab completion motivation, but I think in this case most users should type `next_<tab>` and find their option. I'm very biased by my experience as a user here, but I think the flowing grammatical order makes more sense to me.

Reply:
I agree on this one with @bmorris3, one usually looks for either primary or secondaries rather than just any eclipse types (though with @eteq's suggestion, it would have been easier to have a convenience function of ``next_eclipse()`` that includes both primaries and secondaries.